rtt_rosparam/src/rtt_rosparam_service.cpp

```diff
@@ -239,6 +247,43 @@ class ROSParamService: public RTT::Service
   bool setParamComponentPrivate(const std::string &name) { return set(name, COMPONENT_PRIVATE); }
   bool setParamComponentRelative(const std::string &name) { return set(name, COMPONENT_RELATIVE); }
   bool setParamComponentAbsolute(const std::string &name) { return set(name, COMPONENT_ABSOLUTE); }
+
+  /**
+   * Adds a ROS parameter of any type as a property to this bag.
+   * A Property is created which causes contents of the
+   * property always to be in sync
+   * with the contents of the ROS parameter.
+   * @param name The name of this property
+   * @return the Property object by reference, which you can further query or document.
+   */
+  template<typename T, ResolutionPolicy P>
+  Property<T>& addRosParamProperty( const std::string& name) {
+  const std::string resolved_name = resolvedName(name,P);
+  typename internal::AssignableDataSource<T>::shared_ptr datasource( new RTT::rosparam::RosParamDataSource<T>(resolved_name) );
+    // if (this->getOwner() == nullptr) {
+    //   return nullptr;
+    // }
+    // if (this->getOwner()->properties() == nullptr) {
+    //   return nullptr;
+    // }
+    Property<T>* p = new Property<T>(name,"", datasource);
+    p->doc("Property generated by rosparam attached to parameter: " + resolved_name);
+    this->getOwner()->properties()->ownProperty( p );
+    return *p;
+  }
+
+  /**
+   * An interface ready version of the addRosParamProperty() member function.
+   * The main difference is that it returns the Property instead of a reference
+   * to the Property.
+   * @param name The name of this property
+   * @return the Value o the Property.
+   */
+  template<typename T, ResolutionPolicy P>
+  RTT::Property<T> addRosParamPropertyIface( const std::string& name) {
+    return addRosParamProperty<T, P>(name);;
+  }
```

Comment:
Why is this needed? It does not make a difference for the caller. Why would you want to enforce that a copy is returned?